modules/t/interval_tree_mutable.t

```diff
@@ -0,0 +1,146 @@
+$tree->insert(make_interval(50, 150, 'data1'));
+$tree->insert(make_interval(75, 200, 'data2'));
+is($tree->size(), 2, 'Tree size');
+$search_result = $tree->search(50, 100);
+is(scalar @{$search_result}, 2, 'Number of search results');
+is($search_result->[0]->data, 'data1', 'Correct inclusion');
+is($search_result->[1]->data, 'data2', 'Correct inclusion');
+$search_result = $tree->search(0, 50);
+is(scalar @{$search_result}, 1, 'Number of search results');
+is($search_result->[0]->data, 'data1', 'Correct inclusion');
+$search_result = $tree->search(200, 300);
+is(scalar @{$search_result}, 1, 'Number of search results');
+is($search_result->[0]->data, 'data2', 'Correct inclusion');
+
+$tree = Bio::EnsEMBL::Utils::Tree::Interval::Mutable->new();
+ok(!$tree->remove(make_interval(50, 100)), 'Delete from empty tree');
+
+$tree = Bio::EnsEMBL::Utils::Tree::Interval::Mutable->new();
+$tree->insert(make_interval(75, 150, 'data'));
+is($tree->size(), 1, 'Tree size');
+ok($tree->remove(make_interval(75, 150)), 'Delete from non-empty tree');
+ok(!$tree->size(), 'Tree size');
+ok(!$tree->root, 'Root deleted');
+
+$tree = Bio::EnsEMBL::Utils::Tree::Interval::Mutable->new();
+$tree->insert(make_interval(50, 120, 'data1'));
+$tree->insert(make_interval(75, 100, 'data2'));
+$tree->insert(make_interval(75, 200, 'First data to remove'));
+$tree->insert(make_interval(75, 150, 'Second data to remove'));
+is($tree->size(), 4, 'Tree size');
+$search_result = $tree->search(50, 200);
+is(scalar @{$search_result}, 4, 'Number of search results');
+is($search_result->[0]->data, 'data1', 'Search result');
+is($search_result->[1]->data, 'data2', 'Search result');
+is($search_result->[2]->data, 'First data to remove', 'Search result');
+is($search_result->[3]->data, 'Second data to remove', 'Search result');
+ok($tree->remove(make_interval(75, 200)), 'Delete from node with multiple data');
+is($tree->size(), 3, 'Tree size');
+$search_result = $tree->search(50, 200);
+is(scalar @{$search_result}, 3, 'Number of search results');
+is($search_result->[0]->data, 'data1', 'Search result');
+is($search_result->[1]->data, 'data2', 'Search result');
+is($search_result->[2]->data, 'Second data to remove', 'Search result');
+ok($tree->remove(make_interval(75, 150)), 'Delete from node with multiple data');
+is($tree->size(), 2, 'Tree size');
+$search_result = $tree->search(50, 200);
+is(scalar @{$search_result}, 2, 'Number of search results');
+is($search_result->[0]->data, 'data1', 'Search result');
+is($search_result->[1]->data, 'data2', 'Search result');
+
+$tree = Bio::EnsEMBL::Utils::Tree::Interval::Mutable->new();
+map { $tree->insert($_) } @{$intervals};
+is($tree->size(), scalar @{$intervals}, 'Tree size');
+$search_result = $tree->search(121779004, 121779004);
+is(scalar @{$search_result}, 1, 'Number of search results');
+ok($search_result->[0]->start == 121626874 && $search_result->[0]->end == 122092717, 'Query result interval boundaries');
+
```

Comment:
No test that tree->search() outside of interval returns zero results

Reply:
Done.